### ISSUE TITLE
Debugging with node-inspector fails

### ISSUE THREAD

**Issue by mdotasia** [webdriverio/webdriverio#1640]
## The problem

I'm trying to get debugging with node-inspector work as described here: http://webdriver.io/guide/testrunner/debugging.html. As expected I get message `Debugger listening on [::]:5859`. Then I run

```
node-inspector --debug-port 5859 --no-preload
```

and open

```
http://127.0.0.1:8080/?port=5859
```

in Chrome and see this error in command-line:

```
Node Inspector v0.12.8
Visit http://127.0.0.1:8080/?port=5859 to start debugging.
/usr/local/lib/node_modules/node-inspector/lib/InjectorClient.js:111
      cb(error, NM[0].ref);
                     ^

TypeError: Cannot read property 'ref' of undefined
    at InjectorClient.<anonymous> (/usr/local/lib/node_modules/node-inspector/lib/InjectorClient.js:111:22)
    at /usr/local/lib/node_modules/node-inspector/lib/DebuggerClient.js:121:7
    at Object.value (/usr/local/lib/node_modules/node-inspector/lib/callback.js:23:20)
    at Debugger._processResponse (/usr/local/lib/node_modules/node-inspector/lib/debugger.js:95:21)
    at Protocol.execute (_debugger.js:121:14)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:543:20)
```

Uninstall and reinstall node-inspector did not help.
## Environment
- WebdriverIO version: 4.2.16
- Node.js version: 6.7.0
- [Standalone mode or wdio testrunner](http://webdriver.io/guide/getstarted/modes.html): wdio synchronous tests with mocha
- Additional wdio packages used (if applicable): selenium-standalone
- Chrome: Version 53.0.2785.143 (64-bit)
- OS: macOS 10.12
## Link to Selenium/WebdriverIO logs

Logging webdriverio in `error` mode yields empty log.

**Comment by mdotasia** [comment]
This one seems to be the cause: https://github.com/node-inspector/node-inspector/issues/905.

**Comment by mdotasia** [comment]
Wondering if debugging can be rewritten using `node --inspect` instead of `node-inspector`, see https://medium.com/@paul_irish/debugging-node-js-nightlies-with-chrome-devtools-7c4a1b95ae27#.2abd03s4e.

**Comment by Rkaede** [comment]
Until this is fixed a workaround is to use the debugger in Visual Studio Code.

**Comment by mdotasia** [comment]
@Rkaede How does your workaround work?

**Comment by Joopmicroop** [comment]
My steps tried with version:
node: 6.9.1
wdio: 4.6.1
node-inspector: 0.12.8

Had the same issue, also stumbled upon [node-inspector/node-inspector#905](https://github.com/node-inspector/node-inspector/issues/905)
and even commented out the callback as suggested on [node-inspector/issues/905#issuecomment-249361821](https://github.com/node-inspector/node-inspector/issues/905#issuecomment-249361821) that spun up node-inspector correctly. Well sort of, sometimes I had to pause the debugger and let it go again to get it to respond, sometimes it just hangs...
So that wasn't a solution ether.

I also bumped into the suggested solution of using VSC as suggested by [Rkaede](https://github.com/webdriverio/webdriverio/issues/1640#issuecomment-253552549) above. The reference to that suggestion I found on [Debugging wdio project in VSC](http://blog.likewise.org/2017/02/debugging-a-javascript-webdriverio-project-in-vscode/)
This works fine as long as you use the default wdio.conf.js
Now my search will go on tomorrow on how to get it working with a custom conf...
